e2e/steps/basicSteps.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+var actor = require('../core/core')
+
+module.exports = function() {
+    this.setDefaultTimeout(60 * 1000); //max time before callback
+
+    this.Given(/^I open the (.*) page$/, function (pageName, callback) {
+         actor.goToPage(pageName).then(callback, callback)
+    });
+
+    //this should probably be replaced by doing a rest call and setting a coocky
+    this.When(/^I login with (.*)$/, function (userName, callback) {
+         var user = actor.getUser(userName);
+         Promise.all([
```

Comment:
The promises are coming from the protractor-framework. I need to resolve them here to make sure username and password are set before the login button is pressed. 

The cucumber framework needs the callback to be called when a step is done. Or called with whatever error was thrown. If I do not force this to happen in sequence then the next step might trigger and fail before this one was done. 
In case of an error on any of the promises I need to make sure it is returned to cucumber. Otherwise the testrunnen exits and the tests freeze.